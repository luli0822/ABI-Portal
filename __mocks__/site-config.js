module.exports = {
	assetPrefix: '/some-asset-prefix',
};
