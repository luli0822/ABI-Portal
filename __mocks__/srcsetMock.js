module.exports = {
	sources: {
		'992w': 'mock-992w-img-src',
	},
	srcSet: 'mock-src-set',
};
