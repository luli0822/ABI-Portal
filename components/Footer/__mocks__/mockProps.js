import { pageLinkMock, externalLinkMock } from '../../../.storybook/SiteData';

export const navigationEntries = [
	{
		type: 'link',
		link: pageLinkMock,
		text: 'Link 1',
		accessibleText: undefined,
	},
	{
		type: 'link',
		link: externalLinkMock,
		text: 'Link 2',
		accessibleText: undefined,
	},
];

export const defaultProps = {
	navigationEntries,
};
