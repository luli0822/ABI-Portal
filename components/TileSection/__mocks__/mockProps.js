import { imagePNGMock } from '../../../.storybook/SiteData';
// eslint-disable-next-line import/prefer-default-export

export const primaryMedia = {
	primaryMedia: imagePNGMock,
	primaryLinkAccessibleText: `Image Accessible Text`,
};

export const defaultBlocks = [
	{
		contentType: 'Complex Block',
		id: 'tileBlock1',
		title: 'Tile 1',
		body: {
			// NOTE: For richText body, the CustomDoc component will render the content from values:[] field to eliminate the complex mapping for body.content fields to read the content.
			// For paragraph - {type: 'paragraph' , value: ''}, for list items {type: 'list', value: ['', '']}
			values: [
				{
					type: 'paragraph',
					value: 'Tile 1 Body',
				},
			],
			nodeType: 'document',
			data: {},
			content: [
				{
					nodeType: 'paragraph',
					data: {},
					content: [
						{
							nodeType: 'text',
							value: 'Tile 1 Body',
							data: {},
							marks: [],
						},
					],
				},
			],
		},
		...primaryMedia,
	},
	{
		contentType: 'Complex Block',
		id: 'tileBlock2',
		title: 'Tile 2',
		body: {
			// NOTE: For richText body, the CustomDoc component will render the content from values:[] field to eliminate the complex mapping for body.content fields to read the content.
			// For paragraph - {type: 'paragraph' , value: ''}, for list items {type: 'list', value: ['', '']}
			values: [
				{
					type: 'paragraph',
					value: 'Tile 2 Body',
				},
			],
			nodeType: 'document',
			data: {},
			content: [
				{
					nodeType: 'paragraph',
					data: {},
					content: [
						{
							nodeType: 'text',
							value: 'Tile 2 Body',
							data: {},
							marks: [],
						},
					],
				},
			],
		},
		...primaryMedia,
	},
];

export const defaultContent = {
	contentType: 'Block Group',
	title: 'What can we build for you?',
	body: {
		values: [
			{
				type: 'paragraph',
				value: 'From microsites to interactive quizzes - we know digital experience.',
			},
		],
		nodeType: 'document',
		data: {},
		content: [
			{
				nodeType: 'paragraph',
				data: {},
				content: [
					{
						nodeType: 'text',
						value: 'From microsites to interactive quizzes - we know digital experience.',
						data: {},
						marks: [],
					},
				],
			},
		],
	},
};

// Default props without links
export const defaultProps = {
	content: {
		...defaultContent,
		blocks: defaultBlocks,
	},
};
