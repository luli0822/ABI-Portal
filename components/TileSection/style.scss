@import "../../scss/layout-mixins";
@import "../../scss/text-mixins";
@import "../../scss/button-link-mixins";
@import "../../scss/a11y-mixins";

$root-class: "tile-section";

.#{$root-class} {
  // max-width-section contains the background to the section max-width. Not necessary if your background is 100% width
  @include max-width-section();
  // standard section spacing - not always necessary, not always the best way to provide this (i.e. you might prefer a variable to use in padding or margin as necessary per design)
  @include section-bottom-spacing();

  // most designs won't use base-font-size for this, but here's an example of how you might dynamically build paddings and margins
  padding-top: $base-font-size * 2px;
  padding-bottom: $base-font-size * 2px;

  &__content {
    // this container() provides the max-width that constrains the content, as well as padding and margin needed for the rows and outside gutter
    @include container();
  }

  &__heading {
    @include heading-two();
  }

  &__tile-row {
    // make-row provides the margin and padding necessary for a row
    @include make-row();

    padding-top: $grid-gutter-width / 2;
    padding-bottom: $grid-gutter-width / 2;
  }

  &__tile-col {
    // make-col-ready provides the basic attributes needed for a col
    @include make-col-ready();
    // this generates a 6-column wide column for all breakpoints
    @include make-col(6);

    // this provides media-query for the md viewport and up
    @include media-breakpoint-up(md) {
      // this overrides the column size, setting it to 8 columns wide for md viewport and up
      @include make-col(3);
    }
  }

  &__tile {
    outline: 1px solid $secondary-color;
    background-color: $secondary-color;
    color: $white;
    padding: $grid-gutter-width / 2;
    box-shadow: 0 0 5px $secondary-color;
    height: 100%;
  }

  &__tile-heading {
    @include font-size(30);

    font-family: $heading-font;
    padding-bottom: $grid-gutter-width / 2;
    position: relative;
    padding-top: 50px;
    z-index: 2;
  }

  &__tile-img {
    width: 100%;
    height: 100%;
    object-fit: cover;
    position: absolute;
    left: 0;
    right: 0;
    top: 0;
    z-index: -1;
    opacity: 0.5;
    overflow: hidden;
  }

  &__tile-p {
    padding-top: $grid-gutter-width / 2;
    padding-bottom: $grid-gutter-width / 2;
  }

  &__tile-i {
    font-style: italic;
  }

  &__tile-sup {
    top: -0.5px;
    font-size: 75%;
    line-height: 0;
    position: relative;
    vertical-align: baseline;
  }

  &__cta-col {
    @include make-col-ready();

    @include media-breakpoint-down(sm) {
      @include make-col(6);
    }

    @include media-breakpoint-up(md) {
      @include make-col(4);
    }
  }

  &__rich-p {
    @include body();

    margin-bottom: 0;
    padding-top: $grid-gutter-width / 2;
    padding-bottom: $grid-gutter-width / 2;
  }

  &__rich-b {
    font-weight: bold;
  }

  &__rich-i {
    font-style: italic;
  }

  &__rich-li {
    list-style-type: disc;
  }

  &__cta {
    // includes the mixin to make the __cta element use the global cta-button-primary styles
    @include cta-button-primary();

    margin-bottom: $grid-gutter-width;
  }
}
