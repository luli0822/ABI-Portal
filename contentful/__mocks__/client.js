import fetchSiteData from './fetchSiteData.json';
// Refresh mock siteData will disclude about-us
module.exports = {
	getClient: () => {},
	fetchSiteData: () => fetchSiteData,
};
