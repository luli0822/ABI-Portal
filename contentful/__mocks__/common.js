// A method to build Link data structure
export const linkDataMock = (type, linkId = 'link-id-1') => ({
	sys: {
		id: linkId,
		contentType: {
			sys: {
				type: 'Link',
				linkType: 'ContentType',
				id: `${type}`,
			},
		},
	},
	fields: {
		title: {
			'en-CA': 'Link Text',
		},
		accessibleLabel: {
			'en-CA': 'Link accessible Text',
		},
		url: type === 'externalLink' ? { 'en-CA': 'https://loblawdigital.co' } : undefined,
		page:
			type === 'pageLink'
				? {
						'en-CA': {
							sys: {
								type: 'Link',
								linkType: 'Entry',
								id: 'site-home-page',
							},
						},
				  }
				: undefined,
	},
});

// Use this for pageLink structure mock
export const pageLinkMock = linkDataMock('pageLink');

// Use this externalLink structure mock
export const externalLinkMock = linkDataMock('externalLink');

// A method to build Media/image data structure
export const imageDataMock = type => ({
	sys: {
		type: 'Asset',
		id: type === 'image/png' ? 'asset-id-1' : 'asset-id-2',
	},
	fields: {
		title: {
			'en-CA': 'Image Title Text',
		},
		description: {
			'en-CA': 'Image description Text',
		},
		file: {
			'en-CA': {
				url: 'image-file-url.png',
				details: {
					size: 150795,
					image: {
						width: 300,
						height: 300,
					},
				},
				fileName: 'png-example.png',
				contentType: type,
			},
		},
	},
});
