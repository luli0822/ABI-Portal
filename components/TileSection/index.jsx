import React from 'react';
import './style.scss';
import RichText from '../../contentful/RichText';
import ContentfulImage from '../../contentful/ContentfulImage';
import { contentPropType } from '../../contentful/prop-types';
import TextReplacement from '../../contentful/TextReplacement';

function TileSection(props) {
	const { content } = props;
	const rootClass = 'tile-section';
	// links are references to a link object, so only in the fallback lang

	return (
		<section className={rootClass}>
			<div className={`${rootClass}__content`}>
				<h2 className={`${rootClass}__heading`}>{content.title}</h2>
				{content.body && <RichText document={content.body} rootClass={`${rootClass}__rich`} />}
				{content.blocks && (
					<ul className={`${rootClass}__tile-row`}>
						{content.blocks.map(tile => (
							<li key={tile.id} className={`${rootClass}__tile-col`}>
								<div className={`${rootClass}__tile`}>
									<h3 className={`${rootClass}__tile-heading`}>
										{tile.primaryMedia && (
											<ContentfulImage
												asset={tile.primaryMedia}
												className={`${rootClass}__tile-img`}
											/>
										)}
										{tile.title && (
											<TextReplacement
												text={tile.title}
												replacements={{
													Microsites: <span className={`${rootClass}__tile-i`}>Microsites</span>,
												}}
											/>
										)}
									</h3>

									{tile.body && <RichText document={tile.body} rootClass={`${rootClass}__tile`} />}
								</div>
							</li>
						))}
					</ul>
				)}
			</div>
		</section>
	);
}

TileSection.propTypes = {
	content: contentPropType.isRequired,
};

export default TileSection;
