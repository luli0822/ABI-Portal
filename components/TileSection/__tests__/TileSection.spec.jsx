import React from 'react';
import { shallow } from 'enzyme';
import TileSection from '../index';
import { defaultProps } from '../__mocks__/mockProps';

describe('<TileSection />', () => {
	it('renders correctly', () => {
		// https://github.com/facebook/jest/issues/7802
		// using .debug() instead of adding dependency on enzyme-to-json for now
		// eslint-disable-next-line react/jsx-props-no-spreading
		expect(shallow(<TileSection {...defaultProps} />).debug()).toMatchSnapshot();
	});
});
