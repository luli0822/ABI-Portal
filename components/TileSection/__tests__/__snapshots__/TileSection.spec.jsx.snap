// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TileSection /> renders correctly 1`] = `
"<section className=\\"tile-section\\">
  <div className=\\"tile-section__content\\">
    <h2 className=\\"tile-section__heading\\">
      What can we build for you?
    </h2>
    <RichText document={{...}} rootClass=\\"tile-section__rich\\" replacements={{...}} />
    <ul className=\\"tile-section__tile-row\\">
      <li className=\\"tile-section__tile-col\\">
        <div className=\\"tile-section__tile\\">
          <h3 className=\\"tile-section__tile-heading\\">
            <ContentfulImage asset={{...}} className=\\"tile-section__tile-img\\" alt={[undefined]} lazyload={true} cover={false} />
            <TextReplacement text=\\"Tile 1\\" replacements={{...}} />
          </h3>
          <RichText document={{...}} rootClass=\\"tile-section__tile\\" replacements={{...}} />
        </div>
      </li>
      <li className=\\"tile-section__tile-col\\">
        <div className=\\"tile-section__tile\\">
          <h3 className=\\"tile-section__tile-heading\\">
            <ContentfulImage asset={{...}} className=\\"tile-section__tile-img\\" alt={[undefined]} lazyload={true} cover={false} />
            <TextReplacement text=\\"Tile 2\\" replacements={{...}} />
          </h3>
          <RichText document={{...}} rootClass=\\"tile-section__tile\\" replacements={{...}} />
        </div>
      </li>
    </ul>
  </div>
</section>"
`;
